orchestrator/search/retrieval/retriever.py

```diff
@@ -0,0 +1,447 @@
+from abc import ABC, abstractmethod
```

Comment:
Maybe split up into a package with a module for each retriever type, it's a lot of scrolling now :)